cmd/acr/image_functions.go

```diff
@@ -0,0 +1,252 @@
+				tagFilters[repoName] = tagFilters[repoName] + "|" + tagRegex
+			} else {
+				tagFilters[repoName] = tagRegex
+			}
+		}
+	}
+
+	return tagFilters, nil
+}
+
+func getLastTagFromResponse(resultTags *acr.RepositoryTagsType) string {
+	// The lastTag is updated to keep the for loop going.
+	if resultTags.Header == nil {
+		return ""
+	}
+	link := resultTags.Header.Get(headerLink)
+	if len(link) == 0 {
+		return ""
+	}
+	queryString := strings.Split(link, "?")
+	if len(queryString) <= 1 {
+		return ""
+	}
+	queryStringToParse := strings.Split(queryString[1], ">")
+	vals, err := url.ParseQuery(queryStringToParse[0])
+	if err != nil {
+		return ""
+	}
+	return vals.Get("last")
+}
+
+// getManifests gets all the manifests for the command to be executed on. The command will be executed on this manifest if it does not
+// have any tag and does not form part of a manifest list that has tags referencing it. If the purge command is to be executed,
+// the manifest should also not have a tag and not have a subject manifest.
+func getManifests(ctx context.Context, acrClient api.AcrCLIClientInterface, loginURL string, repoName string, dryRun bool, purge bool) (*[]string, error) {
+	lastManifestDigest := ""
+	var manifestsForCommand []string
+	resultManifests, err := acrClient.GetAcrManifests(ctx, repoName, "", lastManifestDigest)
+	if err != nil {
+		if resultManifests != nil && resultManifests.Response.Response != nil && resultManifests.StatusCode == http.StatusNotFound {
+			fmt.Printf("%s repository not found\n", repoName)
+			return &manifestsForCommand, nil
+		}
+		return nil, err
+	}
+
+	// This will act as a set. If a key is present, then the command shouldn't be executed because it is referenced by a multiarch manifest
+	// or the manifest has subjects attached
+	doNotAffect := set.New[string]()
+	var candidates []acr.ManifestAttributesBase
+	for resultManifests != nil && resultManifests.ManifestsAttributes != nil {
+		manifests := *resultManifests.ManifestsAttributes
+		for _, manifest := range manifests {
+			if manifest.Tags != nil {
+				// If a manifest has Tags and its media type supports multiarch manifest, we will
+				// iterate all its dependent manifests and mark them to not have the command execute on them.
+				err = doNotAffectDependantManifests(ctx, manifest, doNotAffect, acrClient, repoName) // **** TODO: CHANGE FUNCTION
+				if err != nil {
+					return nil, err
+				}
+			} else {
+				if purge {
+					// If a manifest does not have Tags and its media type supports subject, we will
+					// check if the subject exists. If so, the manifest is marked not to be affected by the command.
+					candidates, err = getManifestWithoutSubjectToDelete(ctx, manifest, doNotAffect, candidates, acrClient, repoName)
+					if err != nil {
+						return nil, err
+					}
+				} else {
+					if *manifest.MediaType != "application/vnd.oci.image.manifest.v1+json" {
```

Comment:
When I annotated with EOL, it made that identified that referrer as untagged and so it would annotate that (if I had 2 dangling manifests and annotated those, the next check would say there were 4 dangling manifests). This check was to make sure it wasn't annotating an annotation.

Reply:
If that is the case you should be verifying that the image is a referrer specifically with the artifact type you are looking to avoid. OCI manifests can be used for regular images not just referrers